packages/@lwc/engine-core/src/framework/vm.ts

```diff
@@ -739,7 +748,11 @@ export function resetComponentRoot(vm: VM) {
     for (let i = 0, len = children.length; i < len; i++) {
         const child = children[i];
 
-        if (!isNull(child) && !isUndefined(child.elm)) {
+        // VFragments are special; their .elm property does not point to the root element since they have no root,
+        // so we have to clean them up differently.
+        if (!isNull(child) && isVFragment(child)) {
+            removeFragmentChildren(child, vm);
+        } else if (!isNull(child) && !isUndefined(child.elm)) {
             remove(child.elm, renderRoot);
         }
```

Comment:
```suggestion
        if (!isNull(child)) {
          if (isVFragment(child)) {
              removeFragmentChildren(child, vm);
          } else if (!isUndefined(child.elm)) {
              remove(child.elm, renderRoot);
          }
        }
```

Boolean logic can be simplified a bit